### PR TITLE
chore: track terraform module source type in telemetry

### PR DESCRIPTION
**Pull request by hugodutka**
Addresses https://github.com/coder/nexus/issues/141

### CODE REVIEW

**Comment by hugodutka** [comment]
@johnstcn I made changes according to your feedback. Thanks!

File: coderd/telemetry/telemetry_test.go [hunk 1]
```diff
@@ -142,10 +142,89 @@ func TestTelemetry(t *testing.T) {
...
 		})
-		require.Equal(t, modules[0].Source, "921c61d6f3eef5118f3cae658d1518b378c5b02a4955a766c791440894d989c5")
+		require.Equal(t, modules[0].Source, "ed662ec0396db67e77119f14afcb9253574cc925b04a51d4374bcb1eae299f5d")
 		require.Equal(t, modules[0].Version, "92521fc3cbd964bdc9f584a991b89fddaa5754ed1cc96d6d42445338669c1305")
+		require.Equal(t, modules[0].SourceType, telemetry.ModuleSourceTypeHTTP)
 		require.Equal(t, modules[1].Source, "registry.coder.com/terraform/aws")
 		require.Equal(t, modules[1].Version, "1.0.0")
+		require.Equal(t, modules[1].SourceType, telemetry.ModuleSourceTypeCoderRegistry)
+	})
+	t.Run("ModuleSourceType", func(t *testing.T) {
```
Comment (johnstcn) [inline coderd/telemetry/telemetry_test.go:152]: praise: nice test coverage!

**Review by johnstcn — APPROVED** [review]
One last comment but I don't need to re-review.

File: coderd/telemetry/telemetry.go [hunk 2]
```diff
@@ -680,9 +681,95 @@ func shouldSendRawModuleSource(source string) bool {
...
+	}
+	if strings.Contains(source, "registry.terraform.io") {
+		return ModuleSourceTypePublicRegistry
+	}
+	if strings.Contains(source, "app.terraform.io") || strings.Contains(source, "localterraform.com") {
+		return ModuleSourceTypePrivateRegistry
+	}
+	if strings.Contains(source, "registry.coder.com") {
+		return ModuleSourceTypeCoderRegistry
+	}
```
[the remark this answers is not in the document]
Comment (hugodutka) [reply coderd/telemetry/telemetry.go:765]: @johnstcn I wasn't aware that people could access the coder registry that way. How does this work? Can people self-host our registry?

[the remark this answers is not in the document]
Comment (johnstcn) [reply coderd/telemetry/telemetry.go:765]: As I write this, I realise that prefix-checking won't even help here unless we attempt to parse the module as an URL and extract only the hostname.

In any case, I don't see any reason that someone sufficiently motiviated couldn't do this if they controlled the domain `mydomain.foo.bar` and used something like outsiders/citizen to self-host a registry.

It's more of a contrived hypothetical though and I'm not going to block on this!

[the remark this answers is not in the document]
Comment (hugodutka) [reply coderd/telemetry/telemetry.go:765]: I think it's acceptable to misclassify a small percentage of modules. With telemetry, we’re more focused on aggregates, and cases like this are unlikely to affect the overall picture.

**Pull request merged**